tests/tests/shader/workgroup_size_overrides.rs

```diff
@@ -0,0 +1,103 @@
+use std::mem::size_of_val;
+use wgpu::util::DeviceExt;
+use wgpu::{BufferDescriptor, BufferUsages, Maintain, MapMode};
+use wgpu_test::{gpu_test, GpuTestConfiguration, TestParameters, TestingContext};
+
+const SHADER: &str = r#"
+    override n = 3;
+
+    @group(0) @binding(0)
+    var<storage, read_write> output: array<u32>;
+
+    @compute @workgroup_size(n - 2)
+    fn main(@builtin(local_invocation_index) lii: u32) {
+        output[lii] = lii + 2;
+    }
+"#;
+
+#[gpu_test]
+static WORKGROUP_SIZE_OVERRIDES: GpuTestConfiguration = GpuTestConfiguration::new()
+    .parameters(TestParameters::default().limits(wgpu::Limits::default()))
+    .run_async(move |ctx| async move {
+        workgroup_size_overrides(&ctx, 0, &[2, 0, 0]).await;
+        workgroup_size_overrides(&ctx, 4, &[2, 3, 0]).await;
+        // Expected to fail during pipeline creation:
+        //workgroup_size_overrides(&ctx, 1, &[0, 0, 0]).await;
```

Comment:
If you pass in a should_fail boolean, you can than wrap pipeline creation in a `fail_if` (https://github.com/gfx-rs/wgpu/blob/trunk/tests/src/lib.rs#L73-L84) with that boolean and you'll be able to ensure it properly fails!

Reply:
oh, I knew I had seen that method somewhere